supafunc/_async/functions_client.py

```diff
@@ -0,0 +1,91 @@
+from typing import Any, Dict, Literal, Optional, Union
+
+from httpx import HTTPError, Response
+
+from ..errors import FunctionsHttpError, FunctionsRelayError
+from ..utils import AsyncClient, __version__
+
+
+class AsyncFunctionsClient:
+    def __init__(self, url: str, headers: Dict):
+        self.url = url
+        self.headers = {
+            "User-Agent": f"supabase-py/functions-py v{__version__}",
+            **headers,
+        }
+
+    async def _request(
+        self,
+        method: Literal["GET", "OPTIONS", "HEAD", "POST", "PUT", "PATCH", "DELETE"],
+        url: str,
+        headers: Union[Dict[str, str], None] = None,
+        json: Optional[dict[Any, Any]] = None,
```

Comment:
Tests are failing because we've used the builtin `dict` instead of `typing.Dict` here; can also fix by adding a `from __future import annotations` at the top of this file

Reply:
I was so confused as to why this was failing in the CI but not when I run the tests locally. Thank you for pointing this out.